lib/models/users/matrix.js

```diff
@@ -83,13 +83,17 @@ MatrixUser.prototype.serialize = function() {
  */
 MatrixUser.prototype.escapeUserId = function() {
     // NOTE: Currently Matrix accepts / in the userId, although going forward it will be removed.
-    // NOTE: We also allow uppercase for the time being.
-    const badChars = new Set(this.localpart.replace(/([A-Z]|[a-z]|[0-9]|-|\.|=|_)+/g, ""));
+    const badChars = new Set(this.localpart.replace(/([a-z]|[0-9]|-|\.|=|_)+/g, ""));
     let res = this.localpart;
     badChars.forEach((c) => {
-        const hex = c.charCodeAt(0).toString(16).toLowerCase();
+        const code = c.charCodeAt(0);
+        const hex = code.toString(16).toLowerCase();
+        if (code < 65 || code > 90) {
+            // Alpha codes do not need escaping.
```

Comment:
... but should be lowercased.

Reply:
are we lowercasing them rather than escaping them?

Reply:
I don't see why not. User IDs are semi-opaque anyways, so Bob and bob are just going to have to get along.

Reply:
I can see that causing lots of unexpected conflicts :c

Reply:
tbh I see anything we do here as creating unexpected conflicts. See also: IRC bridge creating duplicate users, resulting in the bridge doing it's own escaping.

Maybe we just keep uppercase as allowed (sorry) and never visit this code again.